File: api/command.rb
require 'mongo'

mongo_client = Mongo::Client.new(ENV['MONGO_URI'], { server_api: { version: '1' } })

def info_body(temperature, humidity, ppm, created_at)
  s = []
  s << "🌡️ Temperature: #{temperature.round(2)}C"
  s << "🚰 Humidity: #{humidity.round(2)}%"
  s << "🦠 PPM: #{ppm.round(2)}ppm"
  s << "🕓 registered at #{created_at.strftime("%H:%M")}"
  return s.join("\n")
end

Handler = Proc.new do |request, response|
  case request.query['command']
  when '/meteo'
    collection = mongo_client['measurements']
    data = collection.find.sort({ _id: -1 }).first
    if data
      response.status = 200
      response.body = info_body(data['temperature'], data['humidity'], data['ppm'], data['_created_at'])
    else
      response.status = 404
      response.body = 'No measurements found'
    end
  else
    response.status = 400
    response.body = 'Bad request'
  end
end

File: api/command.js
const { MongoClient, ServerApiVersion } = require('mongodb')

const client = new MongoClient(process.env.MONGO_URI, {
  useNewUrlParser: true,
  useUnifiedTopology: true,
  serverApi: ServerApiVersion.v1
})

const collection = client.db().collection('measurements')

const infoBody = (temperature, humidity, ppm, createdAt) => (
  [
    `🌡️ Temperature: ${temperature.toFixed(2)}C`,
    `🚰 Humidity: ${humidity.toFixed(2)}%`,
    `🦠 PPM: ${ppm.toFixed(2)}ppm`,
    `registered at ${createdAt.getHours()}:${createdAt.getMinutes()}`
  ].join('\n')
)

module.exports = async (req, res) => {
  // console.log('PIOZ DEBUG OLD STYLE')
  // console.log(req)
  // console.log(req.query)
  console.log(req.body)
  const command = req.query.command

  switch (command) {
    case '/meteo':
      try {
        await client.connect()
        const lastMeasure = await collection.findOne({}, { sort: { _id: -1 } })

        if (lastMeasure) {
          const responseMessage = infoBody(
            lastMeasure.temperature,
            lastMeasure.humidity,
            lastMeasure.ppm,
            lastMeasure._created_at
          )
          res.status(200).send(responseMessage)
        } else {
          res.status(404).send('No data found')
        }
      } catch (error) {
        res.status(500).send('Internal Server Error')
        console.error(error)
      }
      break
    default:
      res.status(400).send('Bad request')
  }
}
